### DISCUSSION TITLE
the demo pod that  validates aad AzureIdentityBinding does not support  AzureCHinaCloud

### DISCUSSION THREAD

**Discussion by Abbott-fu** [Azure/aad-pod-identity#1288] [Q&A] [answered]
In this  article : https://azure.github.io/aad-pod-identity/docs/demo/standard_walkthrough/

at the step : Deploy a pod that validates the functionality 

The demo pod does not support AzureChinaCLoud, when I deploy the pod to AzureCHinaCloud, it sends requests to global Azure . errors like below occur:
E0621 02:36:25.372672       1 main.go:60] failed to refresh the service principal token, error: adal: Refresh request failed. Status Code = '400'. Response body: {"error":"invalid_resource","error_description":"AADSTS500011: The resource principal named **https://management.azure.com/** was not found in the tenant named Microsoft. 

from the source code of the demo pod: https://github.com/Azure/aad-pod-identity/blob/master/cmd/demo/main.go
we can see it uses global Azure endpoint : flag.StringVar(&resourceName, "resource-name", "_https://management.azure.com/_", "The resource name to grant the access token")

Could someone helps add a varible that supports AzureChinaCloud endpoint : https://management.chinacloudapi.cn/ ?

Thanks

**Comment by chewong** [comment, marked as answer]
Have you tried adding the `--resource-name=https://management.chinacloudapi.cn/` flag to the demo pod?

**Reply by Abbott-fu** [reply]
It works , Thanks a lot . yaml is like this:
apiVersion: v1
kind: Pod
metadata:
  name: demo
  labels:
    aadpodidbinding: identityfor0530
spec:
  containers:
  - name: demo
    image: mcr.microsoft.com/oss/azure/aad-pod-identity/demo:v1.8.9
    args:
      - --subscription-id=${SUBSCRIPTION_ID}
      - --resource-group=${IDENTITY_RESOURCE_GROUP}
      - --identity-client-id=${IDENTITY_CLIENT_ID}
      **- --resource-name=https://management.chinacloudapi.cn**
  nodeSelector:
    kubernetes.io/os: linux
